client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientListenerLeakTest.java

```diff
@@ -38,7 +41,10 @@ public void call() {
             eventListener.expectOnlyCreatedEvent(1, cache(cacheName));
          }
       });
-      detectThreadLeaks(".*Client-Listener-ClientListenerLeakTest-.*");
+      WaitFor
+              .condition(() -> listThreadsMatchingRegexp(".*Client-Listener-ClientListenerLeakTest-.*").size() == 0)
+              .atMost(10, TimeUnit.MILLISECONDS)
+              .await();
```

Comment:
I believe so...

Those threads are killed [here](https://github.com/infinispan/infinispan/blob/b4f09f9a59dec2d9ee402b857e61d4d280d16e91/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/event/ClientListenerNotifier.java#L216). Occasionally it might happen that the scheduler doesn't have enough time to clean everything out . Adding this small amount of sleep made it stable on my local environment.

Alternative idea would be to wait until all threads are really dead in the `ClientListenerNotifier#stop` method (but I'm not sure if this is a good idea).